src/pages/workspace/workflows/ToggleSettingsOptionRow.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import type {ViewStyle} from 'react-native';
+import type {SvgProps} from 'react-native-svg';
+import Icon from '@components/Icon';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+
+type OptionType = {
+    icon: React.FC<SvgProps>;
+    title: string;
+    subtitle: string;
+    hasBeenToggled: boolean;
+    onToggle: (isEnabled: boolean) => void;
+    subMenuItems?: React.ReactNode;
+    isEndOptionRow?: boolean;
+};
+
+function ToggleSettingOptionRow({icon, title, subtitle, onToggle, subMenuItems, isEndOptionRow, hasBeenToggled}: OptionType) {
+    const [isEnabled, setIsEnabled] = useState(false);
+    const styles = useThemeStyles();
+
+    const toggleSwitch = () => {
+        setIsEnabled((previousState) => !previousState);
+        onToggle(!isEnabled);
```

Comment:
Is `previousState` the same as `isEnabled`?

Reply:
Good catch, shouldn't need it.